src/Symfony/Security/State/AccessCheckerProvider.php

```diff
@@ -84,10 +84,14 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
             ];
         }
 
+        if ('pre_read' === $this->event && $this->resourceAccessChecker instanceof ObjectVariableCheckerInterface && $this->resourceAccessChecker->usesObjectVariable($isGranted, $resourceAccessCheckerContext)) {
```

Comment:
```suggestion
        if ($preRead && $this->resourceAccessChecker instanceof ObjectVariableCheckerInterface && $this->resourceAccessChecker->usesObjectVariable($isGranted, $resourceAccessCheckerContext)) {
```

Reply:
Also, there is likely a missing test, because the missing use statement hasn't been caught.

Reply:
it did fail on behat :)